File: backend/app/schemas/task.py
from datetime import datetime
from pydantic import BaseModel
from app.models import Priority


class TaskCreate(BaseModel):
    title: str
    description: str | None = None
    priority: Priority = Priority.medium
    due_date: datetime | None = None
    reminder_at: datetime | None = None
    recurrence_cron: str | None = None
    estimated_minutes: int | None = None
    goal_id: int | None = None
    list_name: str | None = None
    parent_list_name: str | None = None


class TaskUpdate(TaskCreate):
    title: str | None = None
    completed: bool | None = None


class TaskRead(TaskCreate):
    id: int
    completed: bool
    completed_at: datetime | None = None
    created_at: datetime
    updated_at: datetime
    is_habit: bool
    goal_id: int | None = None
    external_key: str | None = None

    model_config = {"from_attributes": True}

File: frontend/src/types/task.ts
export type Priority = "high" | "medium" | "low";

export interface Task {
  id: number;
  title: string;
  description?: string;
  priority: Priority;
  due_date?: string;
  reminder_at?: string;
  recurrence_cron?: string;
  estimated_minutes?: number | null;
  list_name?: string | null;
  completed: boolean;
  goal_id?: number | null;
  created_at: string;
  updated_at: string;
}

export interface TaskCreate {
  title: string;
  description?: string;
  priority?: Priority;
  due_date?: string;
  reminder_at?: string;
  recurrence_cron?: string;
  estimated_minutes?: number | null;
  list_name?: string | null;
  goal_id?: number | null;
}

export interface AgendaItem {
  id: string;
  title: string;
  time: string | null;
  priority: Priority | null;
  isEvent: boolean;
  completed: boolean;
  taskId?: number;
  googleId?: string;
  isBlock?: boolean;
  conflict_with?: string | null;
  blockId?: number;
}

export interface CalendarEvent {
  google_id: string;
  title: string;
  start_dt: string | null;
  end_dt: string | null;
  all_day: boolean;
  start_date: string | null;
  done: boolean;
}
